samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -171,12 +186,11 @@ String readProcessorData(String fullPath) {
     String processorPath = keyBuilder.getProcessorsPath();
     List<String> processorIds = new ArrayList<>(znodeIds.size());
     if (znodeIds.size() > 0) {
-
       for (String child : znodeIds) {
         String fullPath = String.format("%s/%s", processorPath, child);
         processorIds.add(new ProcessorData(readProcessorData(fullPath)).getProcessorId());
       }
-
+      Collections.sort(processorIds);
```

Comment:
Any reason why we are sorting this processorIds here?  Looks like previously we weren't and now we do.